### DISCUSSION TITLE
early termination with variable assignment

### DISCUSSION THREAD

**Discussion by binnie268** [open-policy-agent/community#521] [OPA and Rego]
I have seen examples of termination on ground values, but how can we do something similar with variable assignments?

For example if I have the below, if value_match returns true, then stop iteration of data.variables and set the number and resolvecount 's value is set.
```
resolvecount(c) = value {
	value := to_number(ref(c.defaultValue, c.reference))
}

ref(x, y) = v {
	p := resolvepattern(y)
	some i
	value_match(p, data.variables[i])
	v := data.variables[i].value
} else = v {
	v := x
}

value_match(p, variable) {
	p == variable.key
}

value_match(p, variable) {
   "hello" == "variable.key"
}

```

**Comment by anderseknert** [comment]
There’s an open issue for that [here](https://github.com/open-policy-agent/opa/issues/5170). I too would love to see something like a SQL “limit” function for this. Your best option now is to collect in a comprehension and then extract the first item, but it’ll always realize the full collection.
